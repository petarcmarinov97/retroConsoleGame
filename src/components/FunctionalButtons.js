import React, { Component } from 'react';
import Button from './Button';
import "../styles/OptionalButtons.css";
import { functionalButtonCfg } from '../configs/buttonsCfg';

export default class FunctionalButtons extends Component {
  render() {
    return <div className="optional">
      <>
        {functionalButtonCfg.map((btn, index) => (
          <div
            key={index}>
            <Button
              style={btn.style}
              size={btn.size}
              shape={btn.shape}
            />
            <span>{btn.text}</span>
          </div>
        ))}
      </>
    </div>;
  }
}
