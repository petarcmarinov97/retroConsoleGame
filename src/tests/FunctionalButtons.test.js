import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from "enzyme-to-json";
import FunctionalButtons from '../components/FunctionalButtons';

describe('FunctionalButtons tests', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <FunctionalButtons />);
    })

    describe('Snapshots', () => {
        it("should render correctly", () => {
            expect(shallowToJson(wrapper)).toMatchSnapshot();
        })
    })
})