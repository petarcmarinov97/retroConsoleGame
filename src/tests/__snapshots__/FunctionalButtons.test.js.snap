// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FunctionalButtons tests Snapshots should render correctly 1`] = `
<div
  className="optional"
>
  <div
    key="0"
  >
    <Button
      btnkey="80"
      class="tobeclicked"
      shape="circle"
      size="small"
      style={
        Object {
          "background": "rgb(29 201 22)",
          "border": "0.5px solid black",
          "boxShadow": "rgb(174 221 29) 0px 0px 2px 2px inset",
          "visibility": "unset",
        }
      }
    />
    <span>
      Pause(P)
    </span>
  </div>
  <div
    key="1"
  >
    <Button
      btnkey="83"
      class="tobeclicked"
      shape="circle"
      size="small"
      style={
        Object {
          "background": "#98c317",
          "border": "0.5px solid black",
          "boxShadow": "rgb(148 237 4) 0px 0px 2px 2px inset",
          "visibility": "unset",
        }
      }
    />
    <span>
      Sound(S)
    </span>
  </div>
  <div
    key="2"
  >
    <Button
      btnkey="82"
      class="tobeclicked"
      shape="circle"
      size="small"
      style={
        Object {
          "background": "#d70003",
          "border": "0.5px solid black",
          "boxShadow": "rgb(253 52 52) 0px 0px 2px 2px inset",
          "visibility": "unset",
        }
      }
    />
    <span>
      Reset(R)
    </span>
  </div>
  <div
    key="3"
  >
    <Button
      btnkey="32"
      class="tobeclicked"
      shape="circle"
      style={
        Object {
          "border": "1px solid #000",
          "borderColor": "transparent",
          "boxShadow": "0px 0px 2px 1px rgb(255 255 255 / 60%) inset",
          "height": "80px",
          "visibility": "unset",
          "width": "80px",
        }
      }
    />
    <span>
      Drop(SPACE)
    </span>
  </div>
</div>
`;
